BridgeCareApp/BridgeCare/DataAccessLayer/AttributesByYear.cs

```diff
@@ -0,0 +1,112 @@
+﻿using BridgeCare.ApplicationLog;
+using BridgeCare.Interfaces;
+using BridgeCare.Models;
+using BridgeCare.Utility;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+
+namespace BridgeCare.DataAccessLayer
+{
+    public class AttributesByYear : IAttributesByYear
+    {
+        private readonly BridgeCareContext db;
+
+        public AttributesByYear(BridgeCareContext context)
+        {
+            db = context ?? throw new ArgumentNullException(nameof(context));
+        }
+
+        public List<AttributeByYearModel> GetProjectedAttributes(SimulatedSegmentAddressModel segmentAddressModel, BridgeCareContext db)
+        {
+            if (segmentAddressModel == null)
+                return new List<AttributeByYearModel>();
+
+            var selectStatement = String.Format("SELECT * FROM SIMULATION_{0}_{1} WHERE SectionID={2}",
+                segmentAddressModel.NetworkId, segmentAddressModel.SimulationId, segmentAddressModel.SectionId);
+
+            return GeneralAttributeValueQuery(selectStatement);
+        }
+
+        public List<AttributeByYearModel> GetHistoricalAttributes(SectionModel sectionModel, BridgeCareContext db)
+        {
+            if (sectionModel == null)
```

Comment:
If you are checking for null and you have a no-op outside of the if statement, then the logic is a good candidate to be refactored to the NullObject pattern.

Reply:
Well, if the null goes through then the NullReferenceException happens on the string building of the Select. We could try and catch that.
What is done now is to send back an empty result. This case happens if the web query sends no information about what data was wanted (body) , Really an additional message to tell the  web side what happened is optimal. All three authors have different ways of handling error from the api query in yesterdays standup we agreed we need to make a single paradiegm to handle these and the UI people have to be able to handle the errors. This method I use is a middle ground for now. To handle this I essentially need to move up the error handling task to #1 and work will all parties.

Reply:
Implement Error Handling strategy between backend and UI#71

Reply:
I did, it introduces no-result return values if applied to the api case and if applied to the models can support nullable types in the db maybe more eloquently. The former case if introduced would change the complexion of the api code I have written and that of everyone else. Realizing this I   put up car UI#71. If you think I should I can devise a solution and implemented across the board with consultation with the consumers of the calls with attention to common convention, it may indeed use the NullObject pattern in the strategy but there are other ways and objects that already handle nulls. Returning a null from an api call is possible.

Reply:
Implemented a model validation pattern that overides OnActionExecuting in ActionFilterAttribute, this works with logical tags on the model classes that enforce existence and ranges, perhaps lengths on strings. Note, for many reasons keys such as sectionID, simulationID ...should not be '0' , there are many error conditions that would be hard to check for and they would end up returning that row! right now the code uses a range 1- max...we can get rid of this but if the assumption these keys != 0 then it works well.